lib/matplotlib/offsetbox.py

```diff
@@ -531,18 +557,15 @@ def __init__(self, child, pad=None, draw_frame=False, patch_attrs=None):
         if patch_attrs is not None:
             self.patch.update(patch_attrs)
 
-    def get_extent_offsets(self, renderer):
+    def _get_bbox_and_child_offsets(self, renderer):
         # docstring inherited.
-        dpicor = renderer.points_to_pixels(1.)
-        pad = self.pad * dpicor
-        w, h, xd, yd = self._children[0].get_extent(renderer)
-        return (w + 2 * pad, h + 2 * pad, xd + pad, yd + pad,
-                [(0, 0)])
+        pad = self.pad * renderer.points_to_pixels(1.)
+        return (self._children[0].get_bbox(renderer).padded(pad), [(0, 0)])
```

Comment:
This change probably fixes an issue that not all `Artist`s have a `get_extent` method (I do not recall exactly which, but that was a problem when trying to find a way to test `PaddedBox`).

Reply:
They don't all have a get_bbox either.
One important point is that all artists have a get_window_extent, which gives their absolute bbox in display coords; get_bbox is different, it gives their bbox relative to a notional baseline (and x-baseline), in order to enable e.g. alignment of texts with equi-spaced baselines (typically, for legends).